internal/fpath/path_test.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package fpath
+
+import (
+	"fmt"
+	"path"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStorjURL(t *testing.T) {
+	for i, tt := range []struct {
+		url    string
+		bucket string
+		path   string
+		base   string
+		joint  string
+	}{
+		{
+			url:    "sj://mybucket",
+			bucket: "mybucket",
+			path:   "",
+			base:   "",
+			joint:  "suffix",
+		},
+		{
+			url:    "sj://mybucket/myfile",
+			bucket: "mybucket",
+			path:   "myfile",
+			base:   "myfile",
+			joint:  "myfile/suffix",
+		},
+		{
+			url:    "sj://mybucket/myfolder/myfile",
+			bucket: "mybucket",
+			path:   "myfolder/myfile",
+			base:   "myfile",
+			joint:  "myfolder/myfile/suffix",
+		},
+		{
+			url:    "sj://mybucket///myfolder///myfile",
+			bucket: "mybucket",
+			path:   "myfolder/myfile",
+			base:   "myfile",
+			joint:  "myfolder/myfile/suffix",
+		},
+		{
+			url:    "sj:////mybucket///myfolder///myfile",
+			bucket: "mybucket",
+			path:   "myfolder/myfile",
+			base:   "myfile",
+			joint:  "myfolder/myfile/suffix",
+		},
+	} {
+		errTag := fmt.Sprintf("Test case #%d", i)
+
+		fp, err := New(tt.url)
+		assert.NoError(t, err, errTag)
+
+		assert.Equal(t, false, fp.IsLocal(), errTag)
+		assert.Equal(t, tt.bucket, fp.Bucket(), errTag)
+		assert.Equal(t, tt.path, fp.Path(), errTag)
+		assert.Equal(t, tt.base, fp.Base(), errTag)
+		assert.Equal(t, path.Join(tt.path, "suffix"), fp.Join("suffix").Path(), errTag)
+		assert.Equal(t, tt.url+"/suffix", fp.Join("suffix").String(), errTag)
+		assert.Equal(t, tt.url, fp.String(), errTag)
+	}
+}
+
+func TestInvalidStorjURL(t *testing.T) {
+	for i, tt := range []string{
+		"://",
+		"sj://",
+		"sj:///",
+		"sj://mybucket:8080/",
+		"sj:///mybucket:8080/",
+		"sj:////mybucket:8080/",
+		"s3://mybucket/myfolder/myfile",
+	} {
+		errTag := fmt.Sprintf("Test case #%d", i)
+		_, err := New(tt)
+		assert.Error(t, err, errTag)
+	}
+}
+
+func TestLocalPath(t *testing.T) {
+	for i, tt := range []struct {
+		url  string
+		base string
+	}{
+		{
+			url:  "-",
+			base: "-",
+		},
+		{
+			url:  ".",
+			base: ".",
+		},
+		{
+			url:  "..",
+			base: "..",
+		},
+		{
+			url:  "/a/b/c",
+			base: "c",
```

Comment:
Instead of checking for just the base, check for `bases: []string{"a", "b", "c"}`... i.e. multiple calls with `Folder` and `Base` to check whether we are splitting correctly. Although did we need Folder?

Reply:
We don't define `Folder` method in FPath anymore, because it is not needed.

Reply:
Maybe then just adding checks for different path delimiters at the end of the URL?  i.e. `/`, `\`, `:`, `//`, `\\`, just to check correctness in those cases as well.

Reply:
I'll let @stefanbenten add more tests.